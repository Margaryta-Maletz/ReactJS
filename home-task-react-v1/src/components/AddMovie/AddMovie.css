form {
    display: flex;
    flex-wrap: wrap;
    column-gap: 60px;
    row-gap: 30px;
    width: 100%;
}

.wrapper-add_movie-background {
    display: flex;
    flex-wrap: wrap;
    gap: 15px;
    position: fixed;
    top: 0;
    left: 50%;
    transform: translate(-50%, 0);
    width: 100vw;
    height: 100vh;
    background: rgba(35, 35, 35, 0.918051);
    mix-blend-mode: normal;
    z-index: 5;
}

.wrapper.wrapper-add_movie {
    display: flex;
    position: relative;
    flex-wrap: wrap;
    background-color: black;
    margin: 0 auto;
    padding: 60px;
    width: 976px;
    height: 900px;
    z-index: 10;
}

.add_movie-close {
    position: absolute;
    top: 30px;
    right: 30px;
}

.add_movie-title {
    width: 100%;
    font-family: Montserrat;
    font-style: normal;
    font-weight: 300;
    font-size: 40px;
    line-height: 49px;
    letter-spacing: 1px;
    text-transform: uppercase;
    color: #FFFFFF;
    margin: 0 0 38px;
}

.add_movie-label {
    font-family: Montserrat;
    font-style: normal;
    font-weight: 600;
    font-size: 16px;
    line-height: 20px;
    letter-spacing: 1px;
    text-transform: uppercase;
    color: #F65261;
    mix-blend-mode: normal;
    opacity: 0.8;
    width: 490px;
    margin: 0 0 30px;
    padding: 0;
    display: flex;
    flex-direction: column;
    gap: 13px;
}

.add_movie-input {
    background: rgba(50, 50, 50, 0.948044);
    mix-blend-mode: normal;
    opacity: 0.8;
    border-radius: 4px;
    font-family: Montserrat;
    font-style: normal;
    font-weight: normal;
    font-size: 20px;
    line-height: 28px;
    color: #FFFFFF;
    width: 490px;
    padding: 16px ;
    border: none;
}

.add_movie-second_column {
    width: 270px;
}

.add_movie-select {
    width: 522px;
    padding: 16px ;
}

.add_movie-textarea {
    width: 827px;
    height: 197px;
    resize: none;
}

.add_movie-buttons_block {
    width: 100%;
    display: flex;
    justify-content: flex-end;
}

.add_movie-button {
    width: 180px;
    height: 57px;
    border-radius: 4px;
    padding: 0;
    font-family: Montserrat;
    font-style: normal;
    font-weight: 500;
    font-size: 20px;
    line-height: 24px;
    text-align: center;
    text-transform: uppercase;
}

.add_movie-button-reset {
    margin-right: 13px;
    border: 1.5px solid #F65261;
    box-sizing: border-box;
    color: #F65261;
    background: none;
}

.add_movie-button-submit {
    background: #F65261;
    border: none;
    color: #FFFFFF;
}

.add_movie-error {
    color: white;
    padding: 0;
    font-style: normal;
    font-weight: 400;
    font-size: 10px;
    line-height: 14px;
}
