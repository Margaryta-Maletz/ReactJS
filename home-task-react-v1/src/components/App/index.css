.wrapper {
    width: 1200px;
    box-sizing: border-box;
    padding: 20px 60px;
    display: flex;
}
