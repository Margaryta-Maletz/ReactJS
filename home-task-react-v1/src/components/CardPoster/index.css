.card-poster_wrapper {
    width: 322px;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;

    font-family: Montserrat;
    font-style: normal;
    font-weight: 500;
    color: white;
    mix-blend-mode: normal;
}

.card-poster_title {
    font-size: 18px;
    line-height: 22px;
    opacity: 0.7;
    margin: 22px 0 8px;
    width: 75%;
}

.card-poster_genre {
    font-size: 14px;
    line-height: 17px;
    opacity: 0.5;
    margin-block: 0;
    margin: 0 0 50px;
}

.card-poster_release-date {
    width: 66px;
    height: 26px;
    border: 1px solid #555555;
    box-sizing: border-box;
    border-radius: 4px;
    font-size: 14px;
    line-height: 24px;
    text-align: center;
    margin: 22px 0 8px;
}
