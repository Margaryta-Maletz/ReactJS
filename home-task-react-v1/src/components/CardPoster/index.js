import React from 'react';

export const CardPoster = (props) => {
    return (
        <div className='card-poster_wrapper'>
            <img className='card-poster_image' src={ props.cards.imagePath } alt={ props.cards.imageAlt } />
            <h4 className='card-poster_title'>{ props.cards.title }</h4>
            <p className='card-poster_genre'>{ props.cards.genre }</p>
            <p className='card-poster_release-date'>{ props.cards.releaseDate }</p>
        </div>
    )
}
