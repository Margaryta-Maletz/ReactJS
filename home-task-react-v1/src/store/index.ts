import thunk from 'redux-thunk';
import {configureStore, getDefaultMiddleware, ThunkAction, Action} from '@reduxjs/toolkit';
import rootReducer from './slice';

const middleware = getDefaultMiddleware({
    imitableCheck: false,
    serializableCheck: false,
    thunk: true,
});

export const store = configureStore({
    reducer: rootReducer,
    middleware: middleware,
    devTools: process.env.NODE_ENV !== 'production',
});

/*export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> =
    ThunkAction<
        ReturnType,
        RootState,
        unknown,
        Action<string>
    >;*/
