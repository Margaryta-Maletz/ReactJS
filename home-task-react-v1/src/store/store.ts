import {createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import {configureStore, ThunkAction, Action} from '@reduxjs/toolkit';
import rootReducer from './reducers';

export const store = createStore(
    rootReducer,
    applyMiddleware(thunk)
);

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> =
    ThunkAction<
        ReturnType,
        RootState,
        unknown,
        Action<string>
    >;
