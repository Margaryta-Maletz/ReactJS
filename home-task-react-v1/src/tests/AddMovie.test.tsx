import React from "react";
import { initialState } from '../store/initialState';
import { render } from '@testing-library/react';
import { AddMovie } from '../components/AddMovie/';
import {Provider} from "react-redux";
import {store} from "../store/store";

describe('make snapshot', () => {
    it('renders without movie', () => {
        const { container } = render(
            <Provider store={store}>
                <AddMovie setVisibleAddMovie={(value=false) => {return value}}/>
            </Provider>
        );
        expect(container).toMatchSnapshot();
    });
    it('renders with movie', () => {
        const { container } = render(
            <Provider store={store}>
                <AddMovie movie={initialState.movies[0]} setVisibleAddMovie={(value=false) => {return value}}/>
            </Provider>
        );
        expect(container).toMatchSnapshot();
    });
});
