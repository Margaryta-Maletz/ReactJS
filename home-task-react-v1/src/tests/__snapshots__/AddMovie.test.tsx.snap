// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`make snapshot renders with movie 1`] = `
<div>
  <div
    class="wrapper wrapper-add_movie-background"
  >
    <svg
      fill="none"
      height="18"
      viewBox="0 0 150 18"
      width="150"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7.6 5.86c1.347 0 2.427.407 3.24 1.22.827.813 1.24 2.04 1.24 3.68V17H7.56v-5.46c0-1.253-.447-1.88-1.34-1.88-.507 0-.92.18-1.24.54-.307.347-.46.9-.46 1.66V17H0V6.06h4.3v1.08c.427-.427.92-.747 1.48-.96a5.07 5.07 0 011.82-.32zm18.278 5.66c0 .067-.02.42-.06 1.06h-7.6c.16.413.427.733.8.96.374.213.84.32 1.4.32.48 0 .88-.06 1.2-.18.334-.12.694-.327 1.08-.62l2.36 2.38c-1.066 1.173-2.66 1.76-4.78 1.76-1.32 0-2.48-.24-3.48-.72-1-.493-1.773-1.173-2.32-2.04-.546-.867-.82-1.84-.82-2.92 0-1.093.267-2.067.8-2.92a5.599 5.599 0 012.22-2.02c.947-.48 2.007-.72 3.18-.72 1.107 0 2.114.22 3.02.66a5.123 5.123 0 012.18 1.94c.547.853.82 1.873.82 3.06zm-5.98-2.58c-.466 0-.853.127-1.16.38-.306.253-.506.613-.6 1.08h3.52c-.093-.453-.293-.807-.6-1.06-.306-.267-.693-.4-1.16-.4zm15.467 7.68c-.68.387-1.607.58-2.78.58-1.534 0-2.714-.367-3.54-1.1-.827-.747-1.24-1.86-1.24-3.34v-2.8h-1.5v-3.3h1.5V3.6h4.52v3.06h2.24v3.3h-2.24v2.76c0 .333.086.593.26.78.173.187.4.28.68.28.386 0 .726-.1 1.02-.3l1.08 3.14zm5.824-9.96h2.38v3.3h-2.24V17h-4.52V9.96h-1.5v-3.3h1.5V6.6c0-1.427.426-2.553 1.28-3.38.853-.84 2.053-1.26 3.6-1.26.506 0 1 .047 1.48.14.48.093.88.227 1.2.4l-1.08 3.14c-.374-.173-.714-.26-1.02-.26-.334 0-.6.107-.8.32-.187.2-.28.513-.28.94v.02zm3.64-4.5h4.52V17h-4.52V2.16zm6.519 3.9h4.52V17h-4.52V6.06zm2.26-.88c-.814 0-1.474-.213-1.98-.64a2.13 2.13 0 01-.74-1.66c0-.667.246-1.213.74-1.64.506-.44 1.166-.66 1.98-.66.826 0 1.486.207 1.98.62.493.413.74.947.74 1.6 0 .693-.247 1.267-.74 1.72-.494.44-1.154.66-1.98.66zM64.77 17l-1.5-2.34-1.68 2.34h-4.9l4.14-5.42-4.02-5.52h5.1l1.56 2.32 1.64-2.32h4.74l-4.02 5.32 4.12 5.62h-5.18zm8.65-8.82a3.276 3.276 0 011.48-1.4c.654-.32 1.447-.48 2.38-.48v1.86a3.748 3.748 0 00-.44-.02c-1.04 0-1.86.313-2.46.94-.586.613-.88 1.493-.88 2.64V17h-1.92V6.4h1.84v1.78zm10.67 8.94c-1.054 0-2-.233-2.84-.7a5.17 5.17 0 01-1.98-1.92c-.467-.827-.7-1.76-.7-2.8 0-1.04.233-1.967.7-2.78A4.986 4.986 0 0181.25 7c.84-.467 1.786-.7 2.84-.7 1.053 0 1.993.233 2.82.7a4.852 4.852 0 011.96 1.92c.48.813.72 1.74.72 2.78 0 1.04-.24 1.973-.72 2.8a5.027 5.027 0 01-1.96 1.92c-.827.467-1.767.7-2.82.7zm0-1.68c.68 0 1.286-.153 1.82-.46.546-.32.973-.76 1.28-1.32.306-.573.46-1.227.46-1.96 0-.733-.154-1.38-.46-1.94a3.216 3.216 0 00-1.28-1.32c-.534-.307-1.14-.46-1.82-.46-.68 0-1.294.153-1.84.46-.534.307-.96.747-1.28 1.32-.307.56-.46 1.207-.46 1.94 0 .733.153 1.387.46 1.96.32.56.746 1 1.28 1.32.546.307 1.16.46 1.84.46zm18.035-9.04V17h-1.82v-1.6c-.387.547-.9.973-1.54 1.28a4.8 4.8 0 01-2.06.44c-1.414 0-2.527-.387-3.34-1.16-.814-.787-1.22-1.94-1.22-3.46V6.4h1.92v5.88c0 1.027.246 1.807.74 2.34.493.52 1.2.78 2.12.78 1.013 0 1.813-.307 2.4-.92.586-.613.88-1.48.88-2.6V6.4h1.92zm3.655-4.24h1.92V17h-1.92V2.16zm15.166 9.6c0 .147-.013.34-.04.58h-8.6c.12.933.527 1.687 1.22 2.26.707.56 1.58.84 2.62.84 1.267 0 2.287-.427 3.06-1.28l1.06 1.24c-.48.56-1.08.987-1.8 1.28-.707.293-1.5.44-2.38.44-1.12 0-2.113-.227-2.98-.68a5.053 5.053 0 01-2.02-1.94c-.467-.827-.7-1.76-.7-2.8 0-1.027.227-1.953.68-2.78a4.964 4.964 0 011.9-1.92c.813-.467 1.727-.7 2.74-.7 1.013 0 1.913.233 2.7.7.8.453 1.42 1.093 1.86 1.92.453.827.68 1.773.68 2.84zm-5.24-3.84c-.92 0-1.693.28-2.32.84-.613.56-.973 1.293-1.08 2.2h6.8c-.107-.893-.473-1.62-1.1-2.18-.613-.573-1.38-.86-2.3-.86zm13.866 8.46c-.28.24-.627.427-1.04.56-.413.12-.84.18-1.28.18-1.067 0-1.893-.287-2.48-.86s-.88-1.393-.88-2.46V7.98h-1.8V6.4h1.8V4.08h1.92V6.4h3.04v1.58h-3.04v5.74c0 .573.14 1.013.42 1.32.293.307.707.46 1.24.46.587 0 1.087-.167 1.5-.5l.6 1.38zm8.027 0c-.28.24-.626.427-1.04.56-.413.12-.84.18-1.28.18-1.066 0-1.893-.287-2.48-.86-.586-.573-.88-1.393-.88-2.46V7.98h-1.8V6.4h1.8V4.08h1.92V6.4h3.04v1.58h-3.04v5.74c0 .573.14 1.013.42 1.32.294.307.707.46 1.24.46.587 0 1.087-.167 1.5-.5l.6 1.38zm11.55-4.62c0 .147-.013.34-.04.58h-8.6c.12.933.527 1.687 1.22 2.26.707.56 1.58.84 2.62.84 1.267 0 2.287-.427 3.06-1.28l1.06 1.24c-.48.56-1.08.987-1.8 1.28-.707.293-1.5.44-2.38.44-1.12 0-2.113-.227-2.98-.68a5.053 5.053 0 01-2.02-1.94c-.467-.827-.7-1.76-.7-2.8 0-1.027.227-1.953.68-2.78a4.964 4.964 0 011.9-1.92c.813-.467 1.727-.7 2.74-.7 1.013 0 1.913.233 2.7.7.8.453 1.42 1.093 1.86 1.92.453.827.68 1.773.68 2.84zm-5.24-3.84c-.92 0-1.693.28-2.32.84-.613.56-.973 1.293-1.08 2.2h6.8c-.107-.893-.473-1.62-1.1-2.18-.613-.573-1.38-.86-2.3-.86z"
        fill="#F65261"
      />
    </svg>
    <div
      class="wrapper wrapper-add_movie"
    >
      <div
        class="add_movie-close"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 23 24"
          width="23"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1.47099 2.15426L21.529 22.2122"
            stroke="white"
            stroke-linecap="round"
            stroke-width="2"
          />
          <path
            d="M21.529 2.15426L1.47103 22.2122"
            stroke="white"
            stroke-linecap="round"
            stroke-width="2"
          />
        </svg>
      </div>
      <h2
        class="add_movie-title"
      >
        edit movie
      </h2>
      <form
        action="#"
      >
        <label
          class="add_movie-label"
          for="title"
        >
          title
          <input
            class="add_movie-input"
            name="title"
            placeholder="Title movie"
            type="text"
            value="Fifty Shades Freed"
          />
        </label>
        <label
          class="add_movie-label add_movie-second_column"
          for="release_date"
        >
          release date
          <input
            class="add_movie-input add_movie-second_column"
            name="release_date"
            placeholder="Select Date"
            type="date"
            value="2018-02-07"
          />
        </label>
        <label
          class="add_movie-label"
          for="poster_path"
        >
          movie URL
          <input
            autocomplete="off"
            class="add_movie-input"
            name="poster_path"
            placeholder="https://"
            type="text"
            value="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
          />
        </label>
        <label
          class="add_movie-label add_movie-second_column"
          for="vote_average"
        >
          rating
          <input
            class="add_movie-input add_movie-second_column"
            name="vote_average"
            placeholder="7.8"
            step="0.1"
            type="number"
            value="6.1"
          />
        </label>
        <label
          class="add_movie-label"
          for="genres"
        >
          genre
          <div
            class="css-1pcexqc-container add_movie-select"
          >
            <div
              class="css-bg1rzq-control react-select__control"
            >
              <div
                class="css-1hwfws3 react-select__value-container react-select__value-container--is-multi react-select__value-container--has-value"
              >
                <div
                  class="css-es53b3-multiValue react-select__multi-value"
                >
                  <div
                    class="css-12jo7m5 react-select__multi-value__label"
                  >
                    Drama
                  </div>
                  <div
                    class="css-1alnv5e react-select__multi-value__remove"
                  >
                    <svg
                      aria-hidden="true"
                      class="css-19bqh2r"
                      focusable="false"
                      height="14"
                      viewBox="0 0 20 20"
                      width="14"
                    >
                      <path
                        d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  class="css-1g6gooi"
                >
                  <div
                    class="react-select__input"
                    style="display: inline-block;"
                  >
                    <input
                      aria-autocomplete="list"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      id="react-select-3-input"
                      spellcheck="false"
                      style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="css-1wy0on6 react-select__indicators"
              >
                <div
                  aria-hidden="true"
                  class="css-16pqwjk-indicatorContainer react-select__indicator react-select__clear-indicator"
                >
                  <svg
                    aria-hidden="true"
                    class="css-19bqh2r"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                    />
                  </svg>
                </div>
                <span
                  class="css-bgvzuu-indicatorSeparator react-select__indicator-separator"
                />
                <div
                  aria-hidden="true"
                  class="css-16pqwjk-indicatorContainer react-select__indicator react-select__dropdown-indicator"
                >
                  <svg
                    aria-hidden="true"
                    class="css-19bqh2r"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div>
              <input
                name="genres"
                type="hidden"
                value="Drama"
              />
            </div>
          </div>
        </label>
        <label
          class="add_movie-label add_movie-second_column"
          for="runtime"
        >
          runtime
          <input
            autocomplete="off"
            class="add_movie-input add_movie-second_column"
            name="runtime"
            placeholder="minutes"
            step="5"
            type="number"
            value="106"
          />
        </label>
        <label
          class="add_movie-label"
          for="overview"
        >
          overview
          <textarea
            autocomplete="off"
            class="add_movie-input add_movie-textarea"
            name="overview"
            placeholder="Movie description"
          >
            Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.
          </textarea>
        </label>
        <div
          class="add_movie-buttons_block"
        >
          <input
            class="add_movie-button add_movie-button-reset"
            type="reset"
          />
          <input
            class="add_movie-button add_movie-button-submit"
            type="submit"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`make snapshot renders without movie 1`] = `
<div>
  <div
    class="wrapper wrapper-add_movie-background"
  >
    <svg
      fill="none"
      height="18"
      viewBox="0 0 150 18"
      width="150"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7.6 5.86c1.347 0 2.427.407 3.24 1.22.827.813 1.24 2.04 1.24 3.68V17H7.56v-5.46c0-1.253-.447-1.88-1.34-1.88-.507 0-.92.18-1.24.54-.307.347-.46.9-.46 1.66V17H0V6.06h4.3v1.08c.427-.427.92-.747 1.48-.96a5.07 5.07 0 011.82-.32zm18.278 5.66c0 .067-.02.42-.06 1.06h-7.6c.16.413.427.733.8.96.374.213.84.32 1.4.32.48 0 .88-.06 1.2-.18.334-.12.694-.327 1.08-.62l2.36 2.38c-1.066 1.173-2.66 1.76-4.78 1.76-1.32 0-2.48-.24-3.48-.72-1-.493-1.773-1.173-2.32-2.04-.546-.867-.82-1.84-.82-2.92 0-1.093.267-2.067.8-2.92a5.599 5.599 0 012.22-2.02c.947-.48 2.007-.72 3.18-.72 1.107 0 2.114.22 3.02.66a5.123 5.123 0 012.18 1.94c.547.853.82 1.873.82 3.06zm-5.98-2.58c-.466 0-.853.127-1.16.38-.306.253-.506.613-.6 1.08h3.52c-.093-.453-.293-.807-.6-1.06-.306-.267-.693-.4-1.16-.4zm15.467 7.68c-.68.387-1.607.58-2.78.58-1.534 0-2.714-.367-3.54-1.1-.827-.747-1.24-1.86-1.24-3.34v-2.8h-1.5v-3.3h1.5V3.6h4.52v3.06h2.24v3.3h-2.24v2.76c0 .333.086.593.26.78.173.187.4.28.68.28.386 0 .726-.1 1.02-.3l1.08 3.14zm5.824-9.96h2.38v3.3h-2.24V17h-4.52V9.96h-1.5v-3.3h1.5V6.6c0-1.427.426-2.553 1.28-3.38.853-.84 2.053-1.26 3.6-1.26.506 0 1 .047 1.48.14.48.093.88.227 1.2.4l-1.08 3.14c-.374-.173-.714-.26-1.02-.26-.334 0-.6.107-.8.32-.187.2-.28.513-.28.94v.02zm3.64-4.5h4.52V17h-4.52V2.16zm6.519 3.9h4.52V17h-4.52V6.06zm2.26-.88c-.814 0-1.474-.213-1.98-.64a2.13 2.13 0 01-.74-1.66c0-.667.246-1.213.74-1.64.506-.44 1.166-.66 1.98-.66.826 0 1.486.207 1.98.62.493.413.74.947.74 1.6 0 .693-.247 1.267-.74 1.72-.494.44-1.154.66-1.98.66zM64.77 17l-1.5-2.34-1.68 2.34h-4.9l4.14-5.42-4.02-5.52h5.1l1.56 2.32 1.64-2.32h4.74l-4.02 5.32 4.12 5.62h-5.18zm8.65-8.82a3.276 3.276 0 011.48-1.4c.654-.32 1.447-.48 2.38-.48v1.86a3.748 3.748 0 00-.44-.02c-1.04 0-1.86.313-2.46.94-.586.613-.88 1.493-.88 2.64V17h-1.92V6.4h1.84v1.78zm10.67 8.94c-1.054 0-2-.233-2.84-.7a5.17 5.17 0 01-1.98-1.92c-.467-.827-.7-1.76-.7-2.8 0-1.04.233-1.967.7-2.78A4.986 4.986 0 0181.25 7c.84-.467 1.786-.7 2.84-.7 1.053 0 1.993.233 2.82.7a4.852 4.852 0 011.96 1.92c.48.813.72 1.74.72 2.78 0 1.04-.24 1.973-.72 2.8a5.027 5.027 0 01-1.96 1.92c-.827.467-1.767.7-2.82.7zm0-1.68c.68 0 1.286-.153 1.82-.46.546-.32.973-.76 1.28-1.32.306-.573.46-1.227.46-1.96 0-.733-.154-1.38-.46-1.94a3.216 3.216 0 00-1.28-1.32c-.534-.307-1.14-.46-1.82-.46-.68 0-1.294.153-1.84.46-.534.307-.96.747-1.28 1.32-.307.56-.46 1.207-.46 1.94 0 .733.153 1.387.46 1.96.32.56.746 1 1.28 1.32.546.307 1.16.46 1.84.46zm18.035-9.04V17h-1.82v-1.6c-.387.547-.9.973-1.54 1.28a4.8 4.8 0 01-2.06.44c-1.414 0-2.527-.387-3.34-1.16-.814-.787-1.22-1.94-1.22-3.46V6.4h1.92v5.88c0 1.027.246 1.807.74 2.34.493.52 1.2.78 2.12.78 1.013 0 1.813-.307 2.4-.92.586-.613.88-1.48.88-2.6V6.4h1.92zm3.655-4.24h1.92V17h-1.92V2.16zm15.166 9.6c0 .147-.013.34-.04.58h-8.6c.12.933.527 1.687 1.22 2.26.707.56 1.58.84 2.62.84 1.267 0 2.287-.427 3.06-1.28l1.06 1.24c-.48.56-1.08.987-1.8 1.28-.707.293-1.5.44-2.38.44-1.12 0-2.113-.227-2.98-.68a5.053 5.053 0 01-2.02-1.94c-.467-.827-.7-1.76-.7-2.8 0-1.027.227-1.953.68-2.78a4.964 4.964 0 011.9-1.92c.813-.467 1.727-.7 2.74-.7 1.013 0 1.913.233 2.7.7.8.453 1.42 1.093 1.86 1.92.453.827.68 1.773.68 2.84zm-5.24-3.84c-.92 0-1.693.28-2.32.84-.613.56-.973 1.293-1.08 2.2h6.8c-.107-.893-.473-1.62-1.1-2.18-.613-.573-1.38-.86-2.3-.86zm13.866 8.46c-.28.24-.627.427-1.04.56-.413.12-.84.18-1.28.18-1.067 0-1.893-.287-2.48-.86s-.88-1.393-.88-2.46V7.98h-1.8V6.4h1.8V4.08h1.92V6.4h3.04v1.58h-3.04v5.74c0 .573.14 1.013.42 1.32.293.307.707.46 1.24.46.587 0 1.087-.167 1.5-.5l.6 1.38zm8.027 0c-.28.24-.626.427-1.04.56-.413.12-.84.18-1.28.18-1.066 0-1.893-.287-2.48-.86-.586-.573-.88-1.393-.88-2.46V7.98h-1.8V6.4h1.8V4.08h1.92V6.4h3.04v1.58h-3.04v5.74c0 .573.14 1.013.42 1.32.294.307.707.46 1.24.46.587 0 1.087-.167 1.5-.5l.6 1.38zm11.55-4.62c0 .147-.013.34-.04.58h-8.6c.12.933.527 1.687 1.22 2.26.707.56 1.58.84 2.62.84 1.267 0 2.287-.427 3.06-1.28l1.06 1.24c-.48.56-1.08.987-1.8 1.28-.707.293-1.5.44-2.38.44-1.12 0-2.113-.227-2.98-.68a5.053 5.053 0 01-2.02-1.94c-.467-.827-.7-1.76-.7-2.8 0-1.027.227-1.953.68-2.78a4.964 4.964 0 011.9-1.92c.813-.467 1.727-.7 2.74-.7 1.013 0 1.913.233 2.7.7.8.453 1.42 1.093 1.86 1.92.453.827.68 1.773.68 2.84zm-5.24-3.84c-.92 0-1.693.28-2.32.84-.613.56-.973 1.293-1.08 2.2h6.8c-.107-.893-.473-1.62-1.1-2.18-.613-.573-1.38-.86-2.3-.86z"
        fill="#F65261"
      />
    </svg>
    <div
      class="wrapper wrapper-add_movie"
    >
      <div
        class="add_movie-close"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 23 24"
          width="23"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1.47099 2.15426L21.529 22.2122"
            stroke="white"
            stroke-linecap="round"
            stroke-width="2"
          />
          <path
            d="M21.529 2.15426L1.47103 22.2122"
            stroke="white"
            stroke-linecap="round"
            stroke-width="2"
          />
        </svg>
      </div>
      <h2
        class="add_movie-title"
      >
        add movie
      </h2>
      <form
        action="#"
      >
        <label
          class="add_movie-label"
          for="title"
        >
          title
          <input
            class="add_movie-input"
            name="title"
            placeholder="Title movie"
            type="text"
            value=""
          />
        </label>
        <label
          class="add_movie-label add_movie-second_column"
          for="release_date"
        >
          release date
          <input
            class="add_movie-input add_movie-second_column"
            name="release_date"
            placeholder="Select Date"
            type="date"
            value=""
          />
        </label>
        <label
          class="add_movie-label"
          for="poster_path"
        >
          movie URL
          <input
            autocomplete="off"
            class="add_movie-input"
            name="poster_path"
            placeholder="https://"
            type="text"
            value=""
          />
        </label>
        <label
          class="add_movie-label add_movie-second_column"
          for="vote_average"
        >
          rating
          <input
            class="add_movie-input add_movie-second_column"
            name="vote_average"
            placeholder="7.8"
            step="0.1"
            type="number"
            value="0"
          />
        </label>
        <label
          class="add_movie-label"
          for="genres"
        >
          genre
          <div
            class="css-1pcexqc-container add_movie-select"
          >
            <div
              class="css-bg1rzq-control react-select__control"
            >
              <div
                class="css-1hwfws3 react-select__value-container react-select__value-container--is-multi"
              >
                <div
                  class="css-151xaom-placeholder react-select__placeholder"
                >
                  Select Genre
                </div>
                <div
                  class="css-1g6gooi"
                >
                  <div
                    class="react-select__input"
                    style="display: inline-block;"
                  >
                    <input
                      aria-autocomplete="list"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      id="react-select-2-input"
                      spellcheck="false"
                      style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="css-1wy0on6 react-select__indicators"
              >
                <span
                  class="css-bgvzuu-indicatorSeparator react-select__indicator-separator"
                />
                <div
                  aria-hidden="true"
                  class="css-16pqwjk-indicatorContainer react-select__indicator react-select__dropdown-indicator"
                >
                  <svg
                    aria-hidden="true"
                    class="css-19bqh2r"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div>
              <input
                name="genres"
                type="hidden"
              />
            </div>
          </div>
        </label>
        <label
          class="add_movie-label add_movie-second_column"
          for="runtime"
        >
          runtime
          <input
            autocomplete="off"
            class="add_movie-input add_movie-second_column"
            name="runtime"
            placeholder="minutes"
            step="5"
            type="number"
            value=""
          />
        </label>
        <label
          class="add_movie-label"
          for="overview"
        >
          overview
          <textarea
            autocomplete="off"
            class="add_movie-input add_movie-textarea"
            name="overview"
            placeholder="Movie description"
          />
        </label>
        <div
          class="add_movie-buttons_block"
        >
          <input
            class="add_movie-button add_movie-button-reset"
            type="reset"
          />
          <input
            class="add_movie-button add_movie-button-submit"
            type="submit"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;
